### ISSUE TITLE
Soundflower could not find in sound preference in OS X 10.11 beta 1

### ISSUE THREAD

**Issue by denilson1** [RogueAmoeba/Soundflower#35]
All Soundflower driver is missing after upgrade to OS X 10.11 beta 1

**Comment by mattingalls** [comment]
i haven't installed 10.1,  but they might have disallowed all KEXTs in /System folder.
(you probably need a signed version installed in /Library/Extensions/ -- i have one here you can try: https://github.com/mattingalls/Soundflower/releases)

**Comment by LucaTNT** [comment]
You need to disable kext signing 

```
sudo nvram boot-args="kext-dev-mode=1"
```

If it doesn't work, try disabling rootless as well

```
sudo nvram boot-args="kext-dev-mode=1 rootless=0"
```

It worked for me in the developer previews. 

Luca Zorzi

**Comment by marrie** [comment]
Hello to  all. Will an update to fix this problem be out  before  or maybe a little after el capitan is released?  I'm using the gm and have not tested. I'm using the sound flower release from 2013, the one that used to be up on google codes, not this one so wonder if the fix will still work until an update is released ether on github or Rogue Ameba's site. I'll try the fix until then.

thanks.

**Comment by J-Wick4** [comment]
Hey Team,

Any update on this issue? I'm trying to install 1.6.6b on El Capitan 10.11 using the suggestions here and I can't get it working. Really need this plugin and without it I'm not able to upgrade my system.

Thanks!

**Comment by marrie** [comment]
The suggestion of turning off driver signing for that app does not work? I mean telling apple to ignore it?  I dunno if attachments will go through but here is a script you can run from the terminal.

If that fails here is a [dropbox link.](https://www.dropbox.com/s/xiatevblsr2rn3x/sffix.sh?dl=1 <https://www.dropbox.com/s/xiatevblsr2rn3x/sffix.sh?dl=1)

either that or ask apple to sign sound flower's drivers so it can work in time for el capitan's release. Those drivers have not been signed in a while, if ever, I dn't think.

**Comment by J-Wick4** [comment]
Hi Marrie,

Yes, I used that and re-installed, allowed for unsigned installation, and on reboot was still unable to find SoundFlower in the audio settings. I'm using this on a Mac Mini Server Late 2009. It was working on Yosemite. 

For now, I'm just bridgeding the audio ports with a 3.5mm Mini audio cable. X(

**Comment by marrie** [comment]
I hope they can fix this as soon as possible. I use sound flower for my audio productions and demos. I really don't want to be delayed in those, not for my clients' sakes anyway.

Take care.

**Comment by esaruoho** [comment]
I'm running OS X El Capitan 10.11 Beta (15A278b) and unfortunately unable to use Soundflower (or Jack! :D ) so am very stuck there.

I hope Soundflower can somehow work on El Capitan before El Capitan goes public.

SoundFlowerbed says "Soundflower is Not Installed!" after installing Soundflower.

**Comment by marrie** [comment]
Agreed. I'm in panic mode, because I'm supposed to be producing demo tutorials for the blind regarding El Capitan, and I kind of can't. I'm really hoping there can be a fix.

**Comment by LucaTNT** [comment]
This works for me on the GM: https://lucazorzi.net/media/Soundflower/Soundflower.zip

Extract in /System/Library/Extensions, then

```
sudo chown -R root:wheel /System/Library/Extensions/Soundflower.kext
sudo touch /System/Library/Extensions # Forces a refresh of the kext caches
```

And reboot.

Of course you must have disabled SIP (aka rootless) and allowed unsigned kexts before attempting to do this.

**Comment by marrie** [comment]
I will give it a try, but, in the end, sound flower needs to come up with a fix that a non-computer geek could do.

**Comment by LucaTNT** [comment]
Looks like Rogue Amoeba is working on a similar product:
http://rogueamoeba.com/loopback/
http://weblog.rogueamoeba.com/2015/08/08/beta-test-a-brand-new-app/

**Comment by mattingalls** [comment]
@marrie -- did you try installing a signed copy in /Library/Extensions/ (NOT in /System/.. !)-- i have one here you can try: https://github.com/mattingalls/Soundflower/releases - let me know if it works

**Comment by marrie** [comment]
Actually the copy I downloaded 4 years ago still works in yosemite. just not el capitan. Got it from the google codes project when it was hosted there and have not updated since.  I'll look at this  one though, and I'm using el capitan btw not yosemite for this test.

**Comment by marrie** [comment]
Thanks, it worked beautifully. Now let's hope they fix the one up now, or this loop thing will do what sound flower does.

**Comment by mattingalls** [comment]
great!

**Comment by haleyngonadi** [comment]
Ah, worked for me as well... It kept saying error but somehow it shows up even with that. Thank you!

**Comment by J-Wick4** [comment]
Matt, that worked 100% Excellent job, 1,000,000 beer trucks to your house!!

**Comment by marrie** [comment]
Lol. any way to merge this in to the latest that rogue amiba has? or can that not be done.

Take care.

**Comment by briancalderon** [comment]
Why does the audio get obliterated and then just play some strange beep/buzz/whatever sound when I try to open League of Legends? I can get Soundflower to run—but the audio is squeaky and opening the game breaks the audio output completely.

**Comment by bryanbrey** [comment]
MAHALO FROM HAWAII MATT!!!! IT WORKED!!!
Finally got my SoundPimp.com audio enhancement working...

**Comment by marrie** [comment]
I heard they just released sound flower 2.0 that works on el capitan. Where can this be gotten from? on this github page I hope?

Blessings and happy Sunday

**Comment by mattbisme** [comment]
I can confirm that the signed version by @mattingalls appears to be working on the version of El Capitan released today. Only did a quick trial with Audio Hijack, but so far so good.

**Comment by marrie** [comment]
Yeah I confirmed that in the gm release, can't really test with the real release as I'm in the middle of teaching class right now but hope to upgrade tomorrow with out issues.  I'm excited.

**Comment by r618** [comment]
drivers from @mattingalls 's build & soundflowerbed from e.g. last rogue amoeba release works

**Comment by zarroyorosa** [comment]
Hey Im trying to choose on the sunflower to use my headsets and every time I try to click on it the icon from sound flower disappears

**Comment by marrie** [comment]
Choose it from output menu in system prefs. or your recording ssoftware. don't ever use the app.

Take care.

**Comment by zarroyorosa** [comment]
Thanks @marrie. However, its a disappointment that we can't use sunflower like before.

**Comment by marrie** [comment]
Yeah  you can. I've used sound flower this way for 5 years now.  I never e teh app I jus route my sounds t ot then use line in to listen. it's simple really and very easy and intuative to use.

Take care.

**Comment by zarroyorosa** [comment]
@mattingalls @marrie is there any way I can listen to the audio on my headsets from soundflower and my computer audio from then built in speakers? this is how I used to do it before.

**Comment by UllswaterBen** [comment]
Hello - I'm very new to this but would really appreciate some help.  I am trying to get Soundflower working with El Capitan 10.11.1 Beta (15B22c) - my son uses it to create youtube videos and distraught that it does not work any more!

As I read the comments above I understand that the version by @mattingalls should work - but I'm not smart enough to figure out what to do from the link.

I have downloaded the "Soundflower-master" folder from the link but not sure what to do from there.

Any help greatly appreciated - thank you!

**Comment by marrie** [comment]
Use line in. go to Rogue Amebia's main website and it's there under free software.

**Comment by r618** [comment]
@UllswaterBen

**Comment by r618** [comment]
@UllswaterBen - first of all uninstall existing installation of soundflower - there's an uninstall Automator script named "Uninstall Soundflower.scpt" in /Applications/Soundflower
probably reboot )
@mattingalls build is here: https://github.com/mattingalls/Soundflower/releases
download and install latest - 2.0b2 version -, reboot after this - sf drivers are now installed, but the build does not contain sounflowerbed ( the UI app )
dl and run install form http://rogueamoeba.com/freebies/soundflower/ , but install only soundflowerbed (not soundflower itself)
after running soundflowerbed app you should now have it running as before

**Comment by alexwebs** [comment]
How do i disable SIP (aka rootless) and allowed unsigned kexts before attempting to do this.

**Comment by zarroyorosa** [comment]
You can disable the SIP by following this @alexwebs 
<img width="651" alt="disable_enable rootless" src="https://cloud.githubusercontent.com/assets/14933701/10270025/e62e46f4-6ab3-11e5-9fa2-fbf2e7f337ed.png">

I already did it and it works but you risk yourself of the possibility of malware. Also if by chance later on you want to have the Rootless again just do the same thing but type "csrutil enable"

**Comment by alexwebs** [comment]
do we also have to type this in the terminal ?

sudo chown -R root:wheel /System/Library/Extensions/Soundflower.kext
sudo touch /System/Library/Extensions # Forces a refresh of the kext caches

**Comment by zarroyorosa** [comment]
No just "csrutil disable" that will disable completely the rootless and then after you restart your mac for the second time install sunflower from https://rogueamoeba.com/freebies/soundflower/ and it will work like always.

**Comment by mattbisme** [comment]
But you don't have to do any of that to get Soundflower working. Just uninstall and reinstall the new version: https://github.com/mattingalls/Soundflower/releases/tag/2.0b2

**Comment by alexwebs** [comment]
i followed what you said but its till not working not sure how to get it to
work

**Comment by zarroyorosa** [comment]
Here are the steps completely to disable rootless on mac - http://www.macworld.com/article/2986118/security/how-to-modify-system-integrity-protection-in-el-capitan.html

However, before these steps erased/uninstall sunflower and after you follow these steps re-install from  https://rogueamoeba.com/freebies/soundflower/  like a normal app.

**Comment by worldsfinestdj** [comment]
If anyone is still experiencing this problem and is running El Captain on their Mac OSX please go to the link provided to see how to get sound flower back with no issues at all
https://youtu.be/kyS7UG3wmb0

**Comment by elmagow** [comment]
@r618 Could you please explain how to just install soundflowerbed, after installing the new drivers.
Thanks

**Comment by r618** [comment]
@elmagow just run the original installer and uncheck soundflower :) - you can choose the components to be installed

**Comment by elmagow** [comment]
@r618 Oh thanks! Didn't see the button to personalize the installer.

**Comment by perlus** [comment]
hi all, 
thank you Matt for your signed version.

I am running a beta of 10.11.1 and don't get Soundflower to run properly. The Extension seems to load. Soundflowerdbed loads. But when I change the output from soundflower via soundflowerbed (from release 1.66) to any possible target (such as the display port) Soundflowerbed crashes.

When trying to install 2.02b from github I get the warning, that the .pkg cannot be opened because it is not from a verified developer (I then give it the necessary rights through the security pane in system preferences).

Same symptoms  when the csrutil option enabled or disabled.

Any thoughts? 

And: is it possible to manually set soundflower's output? Like changing some line of code? For the time being I would be absolutely willing to do that, or possibly write a script wich changes the necessary lines.

Thank you.

**Comment by mattingalls** [comment]
i'm working on a new version (written from scratch) of Soundflowerbed.  in the meantime, @perlus you could try creating a "multi output device" in the Audio MIDI Setup utility to map the output of soundflower to another device

**Comment by thank-you-matt** [comment]
Just wanted to send a huge thank you to Matt. I struggled with getting sound flower to work, even trying stuff very foreign to me that was way above my head. Yet your link and dmg worked PERFECTLY right from the start, and was the easiest thing I've installed on his mac. THX A TON. Really appreciated.

**Comment by bothkindsofmusic** [comment]
Thanks Matt - I installed El Capitan and went to record my radio show only to find soundflower didn't work. This was the first hit in my search as to why and your solution saved my butt!

**Comment by fredco** [comment]
[Not to bicker, but this installer is not signed.](http://i.imgur.com/fZxFG8B.png)

If the kext is signed, why not the installer?

**Comment by marrie** [comment]
The one that that I'm using  is. I can't remember the link but it works now under El Capitan. At least the driver is signed.

**Comment by alexwebs** [comment]
which one are you using Sarah?

**Comment by marrie** [comment]
I believe I used this one.

https://github.com/mattingalls/Soundflower/releases https://github.com/mattingalls/Soundflower/releases

Just install and it should work.

Good luck.

**Comment by mattingalls** [comment]
yeah, soundflower itself _is_ signed, but he installer isn't (only reason i used an installer here was to uninstall the old version -- the next version won't use an installer at all)

**Comment by alexwebs** [comment]
Matt Ingalls do you have a time frame when you will release it ?

**Comment by elmagow** [comment]
@perlus I had the same problem.
I fixed it by switching first to: "NONE (OFF)" then to the new output.
Sometimes it's still crashing, but just need to force quit it with activity monitor, relaunch soundflowerbed and that's it.
It's not ideal but it's working for me.

**Comment by perlus** [comment]
Yes, tried that too. But i never got it to work the way I need it. Using the midi setup does the job for me now.

**Comment by derwaldgeist** [comment]
@mattingalls: I've tried the version you provided. At first, it seemed to work perfectly. But then I noticed that Ableton Live crashed the whole system quite often when loading a project. According to the Kernel panic, the problem seems to be related to the Audio driver section (at least this is what I'm reading from the report pasted below). So I uninstalled Soundflower, and the problem did not occur again. I'm not 100% sure if the problem has been caused by Soundflower, but it seems as if it did.

Below an excert of the Kernel panic messages. I could provide the full panic report on request.

```
Sun Oct 25 22:06:04 2015

*** Panic Report ***
panic(cpu 6 caller 0xffffff800d5d6bba): Kernel trap at 0xffffff800d30e05b, type 14=page fault, registers:
CR0: 0x000000008001003b, CR2: 0xffffff9220d67000, CR3: 0x00000000118e2000, CR4: 0x00000000001627e0
RAX: 0xffffff81acd93fe8, RBX: 0x0000000000000002, RCX: 0x0000000000000dfd, RDX: 0x0000000000012000
RSP: 0xffffff91f9dcbd68, RBP: 0xffffff91f9dcbda0, RSI: 0xffffff9220d67000, RDI: 0xffffff81acd9f000
R8:  0x0000000000002400, R9:  0xffffff802f2edeb0, R10: 0x0000000000000001, R11: 0x0000000000004000
R12: 0x000000000000cbfa, R13: 0xffffff802f22b000, R14: 0x0000000000002400, R15: 0xffffff802edd6c00
RFL: 0x0000000000010206, RIP: 0xffffff800d30e05b, CS:  0x0000000000000008, SS:  0x0000000000000000
Fault CR2: 0xffffff9220d67000, Error code: 0x0000000000000000, Fault CPU: 0x6, PL: 0

Backtrace (CPU 6), Frame : Return Address
0xffffff91f9dcb9f0 : 0xffffff800d4e5307 
0xffffff91f9dcba70 : 0xffffff800d5d6bba 
0xffffff91f9dcbc50 : 0xffffff800d5f4313 
0xffffff91f9dcbc70 : 0xffffff800d30e05b 
0xffffff91f9dcbda0 : 0xffffff7f8dcdc939 
0xffffff91f9dcbde0 : 0xffffff7f8dcdc752 
0xffffff91f9dcbe00 : 0xffffff7f8dcdc59a 
0xffffff91f9dcbe70 : 0xffffff7f8dcd7e86 
0xffffff91f9dcbed0 : 0xffffff7f8dcd54e8 
0xffffff91f9dcbf00 : 0xffffff800d51965a 
0xffffff91f9dcbfb0 : 0xffffff800d5d15d7 
      Kernel Extensions in backtrace:
         com.apple.iokit.IOAudioFamily(204.1)[14A1C5BC-E9F7-36E0-8AEE-6D1A4C637885]@0xffffff7f8dccc000->0xffffff7f8dcfcfff
            dependency: com.apple.vecLib.kext(1.2.0)[EE6B880D-C8E0-3B4E-8490-43B68F8D2EC0]@0xffffff7f8dc46000
```

**Comment by mattingalls** [comment]
@derwaldgeist - do you have other audio driver KEXTs installed on your system?  does it start crashing again after re-installing soundflower?  interesting that vecLib dependency is listed with-- soundflower doesn't link against vecLib ( but maybe IOKit does ??)

**Comment by derwaldgeist** [comment]
@mattingalls: Thanks for your quick reply. The only audio driver I have installed besides Soundflower is the Steinberg UR44 driver (I installed this one under Yosemite and it still seems to work under El Capitan). I did not try to re-install Soundflower since I was happy that the crashes went away when I removed it, but I can try it.

It's also interesting that the problem always occurred when I tried to load something from my external hard disk. But that may be just by chance, as I was mainly loading Ableton files during that session. I'm using the Paragon NTFS driver for this drive, since I am sharing it with Windows.

**Comment by derwaldgeist** [comment]
Just read on the Steinberg page that the UR44 drivers are not supported for El Capitan yet. So this might be the problem, although the crashes disappeared after uninstalling Soundflower. But I also had some unexplainable sound problems (Spotify stopped playing once in a while), so I'm trying to revert to Yosemite now. It's only one day since I installed El Capitan, so I don't wait until my data is too much out of sync. If you still need the complete crash log, I can provide this.

**Comment by amatt09** [comment]
I installed the latest beta of soundflower on my mac (10.11.1) then uninstalled it. now kernel_task is running at about 100% and my fans are running at full speed. is there something i can do?

**Comment by anr-ableton** [comment]
the best advice here is 
uninstall any soundflower you have, restart and install iShowU HD. Worked for me
as detailed here:
https://support.shinywhitebox.com/hc/en-us/articles/206333005-Soundflower-and-El-Capitan

clicking on record application or something brought up option box to install soundflower

**Comment by DanielJestrab** [comment]
anr-ableton: Definitely worked for me, thanks for the hint!

**Comment by F1LT3R** [comment]
@marrie's link worked for installing sound flower on el capitan for me: https://github.com/mattingalls/Soundflower/releases/tag/2.0b2

**Comment by smial678** [comment]
Hi there - I am completely out of my depth here, please help if you can. I've been following r618's advice above (Oct 4 2015) and have uninstalled old version, downloaded 2.0b2 version from https://github.com/mattingalls/Soundflower/releases then it says go here  http://rogueamoeba.com/freebies/soundflower/ and install only the soundflowerbed BUT everytime it sends me back to https://github.com/mattingalls/Soundflower where I can't find where to download the soundflowerbed from. Any help will be well appreciated, thanks

**Comment by mattingalls** [comment]
soundflowerbed is no longer supported. 
you can do pretty much anything it does within AudioMIDISetup.  read the info on this page https://github.com/mattingalls/Soundflower/releases/tag/2.0b2 for the most common case.  if you need more assistance send me a direct message.

**Comment by r618** [comment]
@smial678 yeah, they've changed the page meanwhile, plx follow @mattingalls advice
if everything else fails i might dig up the old RA version somewhere maybe

**Comment by smial678** [comment]
Hey thanks SO much I really appreciate the help from both of you @r618 and @mattingalls . I followed what you said and am doing it through AMsetup where I am recording the webinar and everyone elses' audio and using Evernote to record my audio. Woohoo I think I may have used my first @link - I don't know what these are called. I may have mentioned I was hopeless.. Ta!

**Comment by elmagow** [comment]
Rogue released a new software called Loopback. I haven't tested it yet but here is the link:
https://rogueamoeba.com/loopback/

**Comment by honk69** [comment]
Hi,
I tried to uninstall soundflower with the scripts and by hand, but I get an error message after every startup.
"soundflower driver not found! Please install the driver."
![soundflower driver not found please install the driver](https://cloud.githubusercontent.com/assets/17525636/13378422/5ff90d0a-de06-11e5-9628-b24462e69367.jpg)

Any help?

Maybe an app, that wants to use soundflower? I'm not sure, if this is a system message or not or what the symbol stands for…

**Comment by mattingalls** [comment]
sounds like you uninstalled it @honk69 - but whatever app that uses that icon seems to want Soundflower installed :)

**Comment by honk69** [comment]
yes, I thought about that too… but when I install it again (latest 2.0b2 version) the window still appears?!?
so maybe the "whatever" wants the old version.

**Comment by mattingalls** [comment]
well look in your System Preferences>Users->Login Items and see if the unknown app shows up there

**Comment by honk69** [comment]
nothing that seems to have a connection to soundflower…
![bildschirmfoto 2016-02-29 um 15 46 47](https://cloud.githubusercontent.com/assets/17525636/13397840/f9b8f04a-defb-11e5-9a5f-7bada0ebe45c.jpg)

Strange is also, that the window started to appear after I installed the 2.0b2 version and used the uninstall script before. So it appears either with or without soundflower installed now, but never appears before "updating".
I also already deleted the soundflower configuration in apples audio-midi setup

**Comment by mattingalls** [comment]
well the fact that it showed up after you uninstalled the old version probably means some app or extension/plugin wants the old version -- pretty scary!

**Comment by honk69** [comment]
yes, slowly I'm running out of ideas…
But thanks a lot for thinking about my problem mattingalls!!

Bad luck I can't find out anything about the symbol shown with the message… I can't find any app with that symbol on my mac.

**Comment by mattingalls** [comment]
maybe reboot and then when that window appears, look at Activity Monitor to see what's running

**Comment by honk69** [comment]
good idea!
I tried that and made a restart with only Activity Monitor and Console opened, but I can't find anything in both.
In Console, there is not even 1 occurrence of soundflower.

I try 1 verbose and 1 safe restart now…

**Comment by honk69** [comment]
well, nothing to read about soundflower in verbose mode, but the window does not appear after a "safe" startup.
For tonight I have to quit, will look into this tomorrow again.

**Comment by mattingalls** [comment]
you can't search for soundflower -- you have to look for the app or process that's displaying that window

**Comment by honk69** [comment]
yes, you're right, just hoped to find something… but I know, I have to find the process displaying the window somehow at least.

thx again mattingalls

**Comment by ozonostudio** [comment]
I can find the little flower on my menu bar, also I can find it anywhere, I already reinstall 4 times, I restart, reboot, try to find it using alfred, and nothing, how I can change the settings if I can't find it?
